### PR TITLE
fix: head font size and alignment

### PR DESCRIPTION
**Pull request by haxxmaxx**
Making sure that the font size is being applied in the header. Also dealing with word wrap and ellipsing for all cells. Applying the same styling and rules as the native table, where all cells have a cap at 3 lines per cell.

the min col width is still kind of arbitrary and pretty big still. Could improve to make dependent on whether the column is locked or not, and maybe make the more button shrink the label when there is not enough space. But this should be good enough for now

### CODE REVIEW

File: src/table/components/CellTextWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable react/require-default-props */
```
Comment (haxxmaxx) [inline src/table/components/CellTextWrapper.tsx:1]: This component is kind of temporary. I'm thinking this wrap/ellipsis solution should be used in all places, but can't do that yet for scrolling

**Comment by cbt1** [comment]
1. Sort direction (and menu) icon does seem to be a bit miss-placed when using a large font size. Maybe center align along the y-axis?
2. I would have expected the column width to ensure that the whole label is visible? As I have not done any column re-sizing

Skärmavbild 2023-02-15 kl  09 52 02

**Comment by cbt1** [comment]
> > 1. Sort direction (and menu) icon does seem to be a bit miss-placed when using a large font size. Maybe center align along the y-axis?
> >    
> >    2. I would have expected the column width to ensure that the whole label is visible? As I have not done any column re-sizing (EDIT: looks like that feature is not in this branch yet? EDIT2: It is? But I cannot drag to re-size. EDIT3: Nv was using the wrong table.. 😂 )
> 
>     1. That is by design, at least initial design. So the idea is that everything should be as close to the bottom border as Possible. It looks pretty good when you have a smaller font size, but I don't think larger are taken into account, especially since the line-height on a large font size adds some white-space under the text.
> 
>     2. well, it is has the fill setting as default. It takes what space it has and divides in between the column, unless you hit the min size then it wont be any smaller. I know there has been discussions of what the default should be

1. It think that is fine, the example I used is a bit extreme in the font size department.
2. Sounds good to me, as long as we have taken a decision on it.

File: src/table/constants.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export enum ColumnWidthTypes {
   FILL = 'fill',
 }
 
-export const MIN_COLUMN_WIDTH = 120;
+export const MIN_COLUMN_WIDTH = 110;
```
[the remark this answers is not in the document]
Comment (cbt1) [reply src/table/constants.ts:44]: Could possibly be an issue with text ellipsis when you hit the smallest width when a dimension is locked.

Guessing that the ellipsis character is wider then the space available so it won't render. Causing the text to appear "clipped".

This is using a default Sense theme.

Skärmavbild 2023-02-15 kl  10 47 02

**Review by a-m-dev — APPROVED** [review]
✅ LGTM

File: src/table/constants.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/table/constants.ts:44]: yeah, maybe we should bump it slightly

File: src/table/components/CellTextWrapper.tsx (hunk 1 above)
Comment (cbt1) [reply src/table/components/CellTextWrapper.tsx:1]: I think you are going to struggle to get this working perfectly with the virtual scroll table. There you need to calculate the width and height of a cell before it's rendered (Unless we somehow can get around that).

Given a certain width and text, you will need figure the number of lines the browser will use to render the text. Which might very well be vary between browsers as I don't think the all use the same word/line wrap algorithm.

My guess as to what will happen is that the browser might require 2-3 lines of text but our calculation might for example guess that it requires 1-2 lines.

But I think we should give it a try anyway!

File: src/table/constants.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbt1) [reply src/table/constants.ts:44]: Increasing the font size will also make this more likely to occur

File: src/table/components/CellTextWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+interface CellTextProps {
+  children: React.ReactNode;
+  lines?: number;
+}
+
+export default function CellTextWrapper({ children, lines = 3 }: CellTextProps) {
+  const {
+    styling: {
+      head: { fontSize },
```
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/table/components/CellTextWrapper.tsx:15]: I think you are right, if I remember correctly, it is the return of `(value) => value.baseProps` that determines the rerenders or not. If the value is the same (===) then it wont rerender. But there are a bunch of thing we don't care about on `baseProps`. Changing to `(value) => value.baseProps.styling.head.fontSize` should do the same thing

File: src/table/constants.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbt1) [reply src/table/constants.ts:44]: Min width for the text should probably be a dynamic value, based on something like the measure width of `M...`

**Review by cbt1 — APPROVED** [review]
LGTM, I think you can increase the min column width and should be good enough for most scenarios

File: src/table/components/CellTextWrapper.tsx (hunk 1 above)
Comment (haxxmaxx) [reply src/table/components/CellTextWrapper.tsx:1]: Well I mainly want to have the same css for word wrap/ellipsis, except that we need to break words in the scrolling solution. I'll leave this like this for now then and then see what I can do for the scrollign

**Pull request merged**